astropy/stats/funcs.py

```diff
@@ -761,6 +766,8 @@ def median_absolute_deviation(a, axis=None):
     # returns an masked array even if the result should be scalar). (#4658)
     if isinstance(a, np.ma.MaskedArray):
         func = np.ma.median
+    elif ignore_nan:
+        func = np.nanmedian
```

Comment:
Something like this could do the trick:

```
try:
    func = np.nanmedian
except AttributeError:
    func = lambda x: np.median(x[~np.isnan(x)])
```